spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -270,223 +315,465 @@ def __call__(self, _, namespace, values, option_string=None):
-    ordered_args = [(arg, value) for (arg, value) in arguments.ordered_args
-                    if arg not in ['i', 'o', 'type', 'shape', 'dim', 'v']]  # Include only array operations
-    if not ordered_args:
-        printv(parser.error('ERROR: you need to specify an operation to do on the input image'))
 
-    # Case 1: Compute similarity metrics
-    if any(getattr(arguments, similarity_arg) is not None for similarity_arg in ['mi', 'minorm', 'corr']):
-        if arguments.mi is not None:
-            # input 1 = from flag -i --> im
-            # input 2 = from flag -mi
-            im_2 = Image(arguments.mi)
-            compute_similarity(im, im_2, fname_out, metric='mi', metric_full='Mutual information', verbose=verbose)
+def apply_div(current_value, *args):
+    """Implementation for -div."""
+    header, [first, *rest] = get_header_and_values(current_value, args)
+    if header is None:
+        return first / math.prod(rest)
+    else:
+        return Image(
+            first / np.prod(rest, axis=0),
+            hdr=header,
+        )
 
-        elif arguments.minorm is not None:
-            im_2 = Image(arguments.minorm)
-            compute_similarity(im, im_2, fname_out, metric='minorm', metric_full='Normalized Mutual information',
-                               verbose=verbose)
 
-        else:
-            assert arguments.corr is not None
-            # input 1 = from flag -i --> im
-            # input 2 = from flag -mi
-            im_2 = Image(arguments.corr)
-            compute_similarity(im, im_2, fname_out, metric='corr', metric_full='Pearson correlation coefficient',
-                               verbose=verbose)
+def apply_mean(current_value, dim):
+    """Implementation for -mean."""
+    if current_value.data.ndim == 3 and dim == 't':
+        return current_value
+    return Image(
+        np.mean(current_value.data, ('x', 'y', 'z', 't').index(dim)),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_rms(current_value, dim):
+    """Implementation for -rms."""
+    data = current_value.data.astype(float)
+    if current_value.data.ndim == 3 and dim == 't':
+        rms = np.abs(data)
+    else:
+        axis = ('x', 'y', 'z', 't').index(dim)
+        rms = np.sqrt(np.mean(np.square(data), axis))
+    return Image(rms, hdr=current_value.hdr)
+
+
+def apply_std(current_value, dim):
+    """Implementation for -std."""
+    return Image(
+        np.std(current_value.data, ('x', 'y', 'z', 't').index(dim), ddof=1),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_bin(current_value, bin_thr):
+    """Implementation for -bin."""
+    return Image(
+        sct_math.binarize(current_value.data, bin_thr),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_otsu(current_value, nbins):
+    """Implementation for -otsu."""
+    return Image(
+        sct_math.otsu(current_value.data, nbins),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_adap(current_value, block_size, offset):
```

Comment:
One thing I like about this structure is that we can give meaningful names to the arguments (like `block_size` and `offset`), rather than always using `arg[0]` and `arg[1]`.